### PR TITLE
Pin Jasmine to compatible version

### PR DESCRIPTION
**Pull request by symbolist**
- Cherrypicks https://github.com/edx/edx-platform/pull/16521 and https://github.com/edx/edx-platform/pull/17980.
- Installs media codes because HLS video player tests fail if browser does not support h264 playback.
- Updates to Firefox 59 because Firefox 45 crashes randomly on Circle CI when running video player tests with h264 codecs installed.

**JIRA tickets**: https://tasks.opencraft.com/browse/OC-4588

**Testing instructions**: Verify all Jasmine tests pass on Circle CI.

### CODE REVIEW

**Comment by symbolist** [comment]
@rocioar This is ready for your review.

File: .circleci/config.yml
Comment (rocioar) [inline .circleci/config.yml]: @symbolist why were this removed?

**Review by rocioar — CHANGES_REQUESTED** [review]

File: .circleci/config.yml
Comment (symbolist) [reply .circleci/config.yml]: @rocioar The current Circle CI plan has only 25 build hours per month so temporarily disabled them while getting js tests working. Will remove this commit before merging.

File: scripts/circle-ci-configuration.sh
Comment (rocioar) [inline scripts/circle-ci-configuration.sh]: We will have to update this comment too, it still says version 45.

**Review by rocioar — APPROVED** [review]
Once you address the small comment, uncomment the rest of the tests, and squash the commits, the PR can be merged.

**Comment by symbolist** [comment]
@rocioar I have addressed your comments. But strangely the lms js suite is now failing with this error: https://openedx.atlassian.net/browse/LEARNER-2120

**Comment by symbolist** [comment]
@rocioar Looks like it had to do with node_modules caching. I have added one more commit which changes the node modules restoration step to only do a restore if the `package.json` hash matches. So whenever `package.json` changes npm will install from scratch. Can you give this last commit a look?

**Comment by rocioar** [comment]
@symbolist good to merge once you have squashed the commits

**Comment by symbolist** [comment]
@rocioar The cherry-picked commits should maintain their provenance. The other two are also independent changes so leaving the commit structure as is. :)

**Pull request merged**
